File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CustomResult.java
package com.oracle.aidataplatform.dp.model;


/**
 * Result from custom tool execution in MCP format.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CustomResult.Builder.class)

public final class CustomResult  {
    @Deprecated
    @java.beans.ConstructorProperties({"content", "structuredContent", "isError", "toolName"})
    public CustomResult(java.util.List<java.util.Map<String, Object>> content, java.util.Map<String, Object> structuredContent, Boolean isError, String toolName) {
        super();
        this.content = content;
        this.structuredContent = structuredContent;
        this.isError = isError;
        this.toolName = toolName;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * MCP-formatted content blocks
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("content")
        private java.util.List<java.util.Map<String, Object>> content;

                /**
         * MCP-formatted content blocks
         * @param content the value to set
         * @return this builder
         **/
        

        public Builder content(java.util.List<java.util.Map<String, Object>> content) {
        this.content = content;
        return this;
        }
            /**
     * Structured result data
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("structuredContent")
        private java.util.Map<String, Object> structuredContent;

                /**
         * Structured result data
         * @param structuredContent the value to set
         * @return this builder
         **/
        

        public Builder structuredContent(java.util.Map<String, Object> structuredContent) {
        this.structuredContent = structuredContent;
        return this;
        }
            /**
     * Whether the result represents an error
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("isError")
        private Boolean isError;

                /**
         * Whether the result represents an error
         * @param isError the value to set
         * @return this builder
         **/
        

        public Builder isError(Boolean isError) {
        this.isError = isError;
        return this;
        }
            /**
     * Name of the executed tool
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("toolName")
        private String toolName;

                /**
         * Name of the executed tool
         * @param toolName the value to set
         * @return this builder
         **/
        

        public Builder toolName(String toolName) {
        this.toolName = toolName;
        return this;
        }


        public CustomResult build() {
            CustomResult model = new CustomResult(this.content
                    , this.structuredContent
                    , this.isError
                    , this.toolName);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CustomResult model) {
                this.content(model.getContent());
    this.structuredContent(model.getStructuredContent());
    this.isError(model.getIsError());
    this.toolName(model.getToolName());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * MCP-formatted content blocks
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("content")
    private final java.util.List<java.util.Map<String, Object>> content;

        /**
     * MCP-formatted content blocks
     * @return the value
     **/
    
    public java.util.List<java.util.Map<String, Object>> getContent() {
        return content;
    }


        /**
     * Structured result data
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("structuredContent")
    private final java.util.Map<String, Object> structuredContent;

        /**
     * Structured result data
     * @return the value
     **/
    
    public java.util.Map<String, Object> getStructuredContent() {
        return structuredContent;
    }


        /**
     * Whether the result represents an error
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isError")
    private final Boolean isError;

        /**
     * Whether the result represents an error
     * @return the value
     **/
    
    public Boolean getIsError() {
        return isError;
    }


        /**
     * Name of the executed tool
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolName")
    private final String toolName;

        /**
     * Name of the executed tool
     * @return the value
     **/
    
    public String getToolName() {
        return toolName;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CustomResult(");
        sb.append("content=").append(String.valueOf(this.content));
        sb.append(", structuredContent=").append(String.valueOf(this.structuredContent));
        sb.append(", isError=").append(String.valueOf(this.isError));
        sb.append(", toolName=").append(String.valueOf(this.toolName));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CustomResult)) {
            return false;
        }

        CustomResult other = (CustomResult) o;
        return java.util.Objects.equals(this.content, other.content) &&
            java.util.Objects.equals(this.structuredContent, other.structuredContent) &&
            java.util.Objects.equals(this.isError, other.isError) &&
            java.util.Objects.equals(this.toolName, other.toolName);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.content == null ? 43 : this.content.hashCode());
        result = (result * PRIME) + (this.structuredContent == null ? 43 : this.structuredContent.hashCode());
        result = (result * PRIME) + (this.isError == null ? 43 : this.isError.hashCode());
        result = (result * PRIME) + (this.toolName == null ? 43 : this.toolName.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/custom_result.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CustomResult(object):
    """
    Result from custom tool execution in MCP format.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CustomResult object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param content:
            The value to assign to the content property of this CustomResult.
        :type content: list[dict(str, object)]

        :param structured_content:
            The value to assign to the structured_content property of this CustomResult.
        :type structured_content: dict(str, object)

        :param is_error:
            The value to assign to the is_error property of this CustomResult.
        :type is_error: bool

        :param tool_name:
            The value to assign to the tool_name property of this CustomResult.
        :type tool_name: str

        """
        self.swagger_types = {
            'content': 'list[dict(str, object)]',
            'structured_content': 'dict(str, object)',
            'is_error': 'bool',
            'tool_name': 'str'
        }

        self.attribute_map = {
            'content': 'content',
            'structured_content': 'structuredContent',
            'is_error': 'isError',
            'tool_name': 'toolName'
        }

        self._content = None
        self._structured_content = None
        self._is_error = None
        self._tool_name = None

    @property
    def content(self):
        """
        Gets the content of this CustomResult.
        MCP-formatted content blocks


        :return: The content of this CustomResult.
        :rtype: list[dict(str, object)]
        """
        return self._content

    @content.setter
    def content(self, content):
        """
        Sets the content of this CustomResult.
        MCP-formatted content blocks


        :param content: The content of this CustomResult.
        :type: list[dict(str, object)]
        """
        self._content = content

    @property
    def structured_content(self):
        """
        Gets the structured_content of this CustomResult.
        Structured result data


        :return: The structured_content of this CustomResult.
        :rtype: dict(str, object)
        """
        return self._structured_content

    @structured_content.setter
    def structured_content(self, structured_content):
        """
        Sets the structured_content of this CustomResult.
        Structured result data


        :param structured_content: The structured_content of this CustomResult.
        :type: dict(str, object)
        """
        self._structured_content = structured_content

    @property
    def is_error(self):
        """
        Gets the is_error of this CustomResult.
        Whether the result represents an error


        :return: The is_error of this CustomResult.
        :rtype: bool
        """
        return self._is_error

    @is_error.setter
    def is_error(self, is_error):
        """
        Sets the is_error of this CustomResult.
        Whether the result represents an error


        :param is_error: The is_error of this CustomResult.
        :type: bool
        """
        self._is_error = is_error

    @property
    def tool_name(self):
        """
        Gets the tool_name of this CustomResult.
        Name of the executed tool


        :return: The tool_name of this CustomResult.
        :rtype: str
        """
        return self._tool_name

    @tool_name.setter
    def tool_name(self, tool_name):
        """
        Sets the tool_name of this CustomResult.
        Name of the executed tool


        :param tool_name: The tool_name of this CustomResult.
        :type: str
        """
        self._tool_name = tool_name

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
